rectools/models/sasrec.py

```diff
@@ -1066,67 +977,20 @@ def _xavier_normal_init(self) -> None:
                 pass
 
 
-# ####  --------------  SASRec Model  --------------  #### #
-
-
-class SASRecModel(ModelBase):
-    """
-    SASRec model for i2i and u2i recommendations.
-
-    n_blocks: int, default 1
-        Number of transformer blocks.
-    n_heads: int, default 1
-        Number of attention heads.
-    n_factors: int, default 128
-        Latent embeddings size.
-    use_pos_emb: bool, default ``True``
-        If ``True``, adds learnable positional encoding to session item embeddings.
-    dropout_rate: float, default 0.2
-        Probability of a hidden unit to be zeroed.
-    session_max_len: int, default 32
-        Maximum length of user sequence.
-    dataloader_num_workers: int, default 0
-        Number of loader worker processes.
-    batch_size: int, default 128
-        How many samples per batch to load.
-    loss: str, default "softmax"
-        Loss function.
-    lr: float, default 0.01
-        Learning rate.
-    epochs: int, default 3
-        Number of training epochs.
-    verbose: int, default 0
-        Verbosity level.
-    deterministic: bool, default ``False``
-        If ``True``, sets deterministic algorithms for PyTorch operations.
-        Use `pytorch_lightning.seed_everything` together with this parameter to fix the random state.
-    cpu_n_threads: int, default 0
-        Number of threads to use in ranker.
-    trainer: Optional(Trainer), default None
-        Which trainer to use for training.
-        If trainer is None, default pytorch_lightning Trainer is created.
-    item_net_type: Type(ItemNetBase), default `IdEmbeddingsItemNet`
-        Type of network returning item enbeddings.
-    pos_encoding_type: Type(PositionalEncodingBase), default `LearnableInversePositionalEncoding`
-        Type of positional encoding.
-    transformer_layers_type: Type(TransformerLayersBase), default `SasRecTransformerLayers`
-        Type of transformer layers architecture.
-    data_preparator_type: Type(SessionEncoderDataPreparatorBase), default `SasRecDataPreparator`
-        Type of data preparator used for dataset processing and dataloader creation.
-    lightning_module_type: Type(SessionEncoderLightningModuleBase), default `SessionEncoderLightningModule`
-        Type of lightning module defining training procedure.
-    """
+class TransformerModelBase(ModelBase):
+    """TODO"""
```

Comment:
"""Base model for all recommender algorithms that work on transformer architecture (e.g. SASRec, Bert4Rec). To create a custom transformer model it is necessary to inherit from this class and write `self.data_preparator` initialization logic.
"""

Reply:
fixed